starknet_py/net/client_models.py

```diff
@@ -702,3 +704,140 @@ class SignatureOnStateDiff:
     block_number: int
     signature: List[int]
     signature_input: SignatureInput
+
+
+# ------------------------------- Trace API dataclasses -------------------------------
+
+
+class SimulationFlags(str, Enum):
+    SKIP_VALIDATE = "SKIP_VALIDATE"
+    SKIP_FEE_CHARGE = "SKIP_FEE_CHARGE"
+
+
+@dataclass
+class EventContent:
+    """
+    Dataclass representing contents of an event.
+    """
+
+    keys: List[int]
+    data: List[int]
+
+
+class SimulationFlag(Enum):
+    """
+    Enum class representing possible simulation flags for trace API.
+    """
+
+    SKIP_VALIDATE = "SKIP_VALIDATE"
+    SKIP_FEE_CHARGE = "SKIP_FEE_CHARGE"
+
+
+class EntryPointType(Enum):
+    """
+    Enum class representing entry point types.
+    """
+
+    EXTERNAL = "EXTERNAL"
+    L1_HANDLER = "L1_HANDLER"
+    CONSTRUCTOR = "CONSTRUCTOR"
+
+
+class CallType(Enum):
+    """
+    Enum class representing call types.
+    """
+
+    LIBRARY_CALL = "LIBRARY_CALL"
+    CALL = "CALL"
+
+
+@dataclass
+class FunctionInvocation:
+    """
+    Dataclass representing an invocation of a function.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+    contract_address: int
+    entry_point_selector: int
+    calldata: List[int]
+    caller_address: int
+    class_hash: int
+    entry_point_type: EntryPointType
+    call_type: CallType
+    result: List[int]
+    calls: List["FunctionInvocation"]
+    events: List[Event]
+    messages: List[L2toL1Message]
+
+
+@dataclass
+class InvokeTransactionTrace:
+    """
+    Dataclass representing a transaction trace of an INVOKE transaction.
+    """
+
+    validate_invocation: Optional[FunctionInvocation]
+    execute_invocation: Union[
+        FunctionInvocation, dict
+    ]  # dict in case "revert_reason" is returned
```

Comment:
What do you think of creating a dedicated dataclass like `RevertedExecuteInvocation` or `RevertedFunctionInvocation` instead of using dict? It would have a single field `revert_reason`, but the purpose of return type would be clearer and it would be easier to expand error response should it be changed.